src/cpprealm/experimental/managed_numeric.hpp

```diff
@@ -18,34 +18,96 @@ namespace realm::experimental {
+            if (this->should_detect_usage_for_queries) {
+                auto query = internal::bridge::query(this->query->get_table());
+                query.not_equal(this->m_key, (int64_t)rhs);
+                return rbool(std::move(query));
+            }
+            return serialize(value()) != rhs;
+        }
+
+        template<typename T>
+        std::enable_if_t<std::is_integral_v<T>, rbool> operator>(const T& rhs) const noexcept {
+            if (this->should_detect_usage_for_queries) {
+                auto query = internal::bridge::query(this->query->get_table());
+                query.greater(this->m_key, (int64_t)rhs);
+                return rbool(std::move(query));
+            }
+            return serialize(value()) > rhs;
+        }
+
+        template<typename T>
+        std::enable_if_t<std::is_integral_v<T>, rbool> operator<(const T& rhs) const noexcept {
+            if (this->should_detect_usage_for_queries) {
+                auto query = internal::bridge::query(this->query->get_table());
+                query.less(this->m_key, (int64_t)rhs);
+                return rbool(std::move(query));
+            }
+            return serialize(value()) < rhs;
+        }
+
+        template<typename T>
+        std::enable_if_t<std::is_integral_v<T>, rbool> operator>=(const T& rhs) const noexcept {
+            if (this->should_detect_usage_for_queries) {
+                auto query = internal::bridge::query(this->query->get_table());
+                query.greater_equal(this->m_key, (int64_t)rhs);
+                return rbool(std::move(query));
+            }
+            return serialize(value()) >= rhs;
+        }
+
+        template<typename T>
+        std::enable_if_t<std::is_integral_v<T>, rbool> operator<=(const T& rhs) const noexcept {
+            if (this->should_detect_usage_for_queries) {
+                auto query = internal::bridge::query(this->query->get_table());
+                query.less_equal(this->m_key, (int64_t)rhs);
+                return rbool(std::move(query));
+            }
+            return serialize(value()) <= rhs;
+        }
+
+        managed& operator+=(const int64_t& o) {
             auto old_val = m_obj->template get<int64_t>(m_key);
             m_obj->template set<int64_t>(this->m_key, old_val + o);
+            return *this;
         }
-        void operator++() {
+        void operator++(int) {
             auto old_val = m_obj->template get<int64_t>(m_key);
-            m_obj->template set<int64_t>(this->m_key, old_val++);
+            m_obj->template set<int64_t>(this->m_key, old_val + 1);
         }
-        void operator-=(const int64_t& o) {
+        managed& operator-=(const int64_t& o) {
             auto old_val = m_obj->template get<int64_t>(m_key);
             m_obj->template set<int64_t>(this->m_key, old_val - o);
+            return *this;
         }
-        void operator--() {
+        void operator--(int) {
```

Comment:
Why the change? This was for the iterator ifc.

Reply:
I've added back the prefix operators.